examples/recaptcha-basic/src/main.rs

```diff
@@ -1,40 +1,48 @@
 use gloo_console::log;
 use yew::prelude::*;
-use yew_recaptcha_v3::recaptcha::Recaptcha;
+use yew_recaptcha_v3::recaptcha::use_recaptcha;
 
 const RECAPTCHA_SITE_KEY: &str = std::env!("RECAPTCHA_SITE_KEY");
 
 #[function_component(App)]
 fn app_component() -> Html {
-    let on_execute = Box::new(use_state(|| None));
     let last_token = Box::new(use_state(|| None));
+    let on_execute = Box::new(use_state(|| None));
     let on_click = {
         let on_execute = on_execute.clone();
         let last_token = last_token.clone();
         Callback::from(move |_| {
             log!("Button clicked");
-            // Per https://yew.rs/docs/next/concepts/function-components/communication
-            // We need to create a new callback everytime that we want Recaptcha to be executed.
             let last_token = last_token.clone();
+            
+            // setting the on_execute callback will force recaptcha to be recalculated.
             on_execute.set(Some(Callback::from(move |token| {
                 last_token.set(Some(token));
             })));
             ()
         })
     };
+    let counter = use_state(|| 0);
+    let on_counter_click = {
+        let counter = counter.clone();
+        move |_| {
+            let value = *counter + 1;
+            counter.set(value);
+        }
+    };
+
+    // Recaptcha will be called only when on_execute changes.
+    use_recaptcha(RECAPTCHA_SITE_KEY.to_string(), on_execute);
```

Comment:
@griffobeid I refactored your code so that we do not have to keep tabs on the # of times that we have called the API, instead, when the user sets the `on_execute` callback, recaptcha will be recomputed.